pkg/authenticator/config/config.go

```diff
@@ -28,88 +29,109 @@ const (
 	DefaultClientCertPath = "/etc/conjur/ssl/client.pem"
 	DefaultTokenFilePath  = "/run/conjur/access-token"
 
+	DefaultConjurVersion = "5"
+
 	// DefaultTokenRefreshTimeout is the default time the system waits to reauthenticate on error
 	DefaultTokenRefreshTimeout = 6 * time.Minute
 )
 
-// New returns a new authenticator configuration object
+var requiredEnvVariables = []string{
+	"CONJUR_AUTHN_URL",
+	"CONJUR_ACCOUNT",
+	"CONJUR_AUTHN_LOGIN",
+	"MY_POD_NAMESPACE",
+	"MY_POD_NAME",
+}
+
+// NewFromEnvFunc defines the method for creating a new config from the env
+type NewFromEnvFunc func(Options) (*Config, error)
+
+// ReadSSLCertFunc defines the method for reading an SSL Certificate from the env
+type ReadSSLCertFunc func() ([]byte, error)
+
+// NewUserNameFunc defines the method for parsing a username from the env
+type NewUserNameFunc func(string) (*Username, error)
+
+// Options contains the functional options needed for NewFromEnv
+type Options struct {
+	ReadSSLCert ReadSSLCertFunc
+	NewUserName NewUserNameFunc
+}
+
+// DefaultConfigOptions implement the methods defined within config.go
+var DefaultConfigOptions = Options{
+	readSSLCert,
+	NewUsername,
+}
+
+// NewFronEnv returns a config FromEnv with the default functional options
 func NewFromEnv() (*Config, error) {
-	var err error
+	return FromEnv(DefaultConfigOptions)
+}
+
+// FromEnv returns a new authenticator configuration object
+func FromEnv(opts Options) (*Config, error) {
+	var err error = nil
 
 	// Check that required environment variables are set
-	for _, envvar := range []string{
-		"CONJUR_AUTHN_URL",
-		"CONJUR_ACCOUNT",
-		"CONJUR_AUTHN_LOGIN",
-		"MY_POD_NAMESPACE",
-		"MY_POD_NAME",
-	} {
+	for _, envvar := range requiredEnvVariables {
 		if os.Getenv(envvar) == "" {
 			return nil, log.RecordedError(log.CAKC009E, envvar)
 		}
 	}
 
+	config := NewDefaultConfig()
+
 	// Load CA cert
-	caCert, err := readSSLCert()
+	config.SSLCertificate, err = opts.ReadSSLCert()
 	if err != nil {
 		return nil, log.RecordedError(log.CAKC021E, err.Error())
 	}
 
 	// Load configuration from the environment
-	authnURL := os.Getenv("CONJUR_AUTHN_URL")
-	account := os.Getenv("CONJUR_ACCOUNT")
-	authnLogin, err := NewUsername(os.Getenv("CONJUR_AUTHN_LOGIN"))
+	config.Username, err = opts.NewUserName(os.Getenv("CONJUR_AUTHN_LOGIN"))
```

Comment:
Apologies if this comment shows up twice, but I didn't see it showing up in the file diffs:
I believe that the NewUserName call here doesn't need to be mocked for UT. To convince myself, I put back the original call here on line 92, and the UT ran fine.